### PR TITLE
Config rework design doc

### PR DESCRIPTION
**Pull request by crufter**

### CODE REVIEW

File: design/config.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	// Init the config
+	Init(opts ...Option) error
+}
+```
+
+## Changes
+
+- Removed all config implementations (file/memory etc), now we only have 1 very light implementation which uses the `Store` interface
+- Removed `Source`, `Loader`, `Watcher`, `ChangeSet` (auditing) concepts entirely
+- `Values` interface now supports options (needed for secrets and for a `Key` option which will drive namespacing now that store interaction is pushed into the config implementation, this previously happened in the config server handler)
```
Comment (domwong) [inline design/config.md:57]: Can you add a quick example of how you think it'll look to actually use the options for secrets

Comment (crufter) [reply design/config.md:57]: Sure

Comment (crufter) [reply design/config.md:57]: Done

**Review by domwong — APPROVED** [review]
I think this looks good for now. Nice and simple. Moves us away from issues like we had with updating config which then gets overridden by a watcher.

Has room to grow for things like adding change watchers back etc.

File: design/config.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+config.Get("path.to.secret", config.Secret(true))
+```
+
+## Handling encryption keys
+
+### Proposal 1
+
+Keep it simple. Have one key to encrypt all secret values across all namespaces.
+
+### Proposal 2
```
Comment (crufter) [inline design/config.md:78]: Probably not feasible to break the key https://crypto.stackexchange.com/questions/3043/how-much-computing-resource-is-required-to-brute-force-rsa

Comment (crufter) [reply design/config.md:78]: Edit: although this is interesting "There are much faster ways to find out a secret key. In fact there are algorithms with sub-exponential running time for factoring integers."

**Pull request merged**